providers/metrics/common/utils.go

```diff
@@ -0,0 +1,26 @@
+package common
+
+import (
+	"compress/gzip"
+	"io"
+)
+
+// CompressWithGzip takes an io.Reader as input and pipes
+// it through a gzip.Writer returning an io.Reader containing
+// the gzipped data.
+// An error is returned if passing data to the gzip.Writer fails
+func CompressWithGzip(data io.Reader) (io.Reader, error) {
```

Comment:
`CompressWithGzip` should be placed under the pkg package of the erda-infra repo.